gitlab/mixins.py

```diff
@@ -173,7 +172,7 @@ def refresh(self, **kwargs: Any) -> None:
             GitlabGetError: If the server cannot perform the request
         """
         if self._id_attr:
-            path = f"{self.manager.path}/{self.id}"
+            path = f"{self.manager.path}/{self.encoded_id}"
```

Comment:
Ok, maybe I'm starting to see having a strict "only encode once" mechanism will help.

Currently this breaks if the user creates a project lazily and then refreshes it, as it'll be encoded twice.

```
>>> import gitlab
>>> gl = gitlab.Gitlab()
>>> project = gl.projects.get("gitlab-org/gitlab", lazy=True)
>>> project.id
'gitlab-org%2Fgitlab'
>>> project.refresh()
Traceback (most recent call last):
  File "/home/nejc/repos/python-gitlab/gitlab/exceptions.py", line 307, in wrapped_f
    return f(*args, **kwargs)
  File "/home/nejc/repos/python-gitlab/gitlab/mixins.py", line 180, in refresh
    server_data = self.manager.gitlab.http_get(path, **kwargs)
  File "/home/nejc/repos/python-gitlab/gitlab/client.py", line 761, in http_get
    result = self.http_request(
  File "/home/nejc/repos/python-gitlab/gitlab/client.py", line 727, in http_request
    raise gitlab.exceptions.GitlabHttpError(
gitlab.exceptions.GitlabHttpError: 404: 404 Project Not Found
```

Can we continue with this PR and fix these specific paths, and then follow up with a robust check?

Reply:
I added a new commit to this PR to hopefully resolve this issue.